src/lib/createPollModal.ts

```diff
@@ -16,8 +16,11 @@ export async function createPollModal({ id = '', question, persistence, data, mo
 }): Promise<IUIKitModalViewParam> {
     const viewId = id || `create-poll-modal-${uuid()}`;
 
-    const association = new RocketChatAssociationRecord(RocketChatAssociationModel.MISC, viewId);
-    await persistence.createWithAssociation(data, association);
+    const viewAssociation = new RocketChatAssociationRecord(RocketChatAssociationModel.MISC, viewId);
+    await persistence.createWithAssociation(data, viewAssociation);
```

Comment:
This should be `updateByAssociation` with `upsert` set to `true` since `createPollModal` is called everytime the "Add Choice" button is clicked with an updated value of options. Using `createWithAssociation` only persists the first option value (2) since a newer record is not created.

Reply:
Nice catch!!